e2e-test/trace/src/integrationTest/java/com/amazon/dataprepper/integration/trace/EndToEndRawSpanTest.java

```diff
@@ -165,19 +168,27 @@ private List<Map<String, Object>> getSourcesFromSearchHits(final SearchHits sear
             if (source.containsKey(TraceGroup.TRACE_GROUP_NAME_FIELD)) {
                 // Extract and replace traceGroupFields with searchHit fields to unify the representation
                 source.entrySet().removeIf(entry -> entry.getKey().startsWith("traceGroupFields"));
-                final Number statusCode = hit.field(TraceGroup.TRACE_GROUP_STATUS_CODE_FIELD).getValue();
+
+                final DocumentField statusCodeField = hit.field(TraceGroup.TRACE_GROUP_STATUS_CODE_FIELD);
```

Comment:
You should be able to remove these two lines since the `addNonNullField` takes care of this now.

Reply:
Good catch.